src/BackgroundTasks/Bucket.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace ILIAS\BackgroundTasks;
+
+/**
+ * Interface Bucket
+ *
+ * @package ILIAS\BackgroundTasks
+ *
+ *          A Bucket is used to pack several Tasks which have to be done one after the other and is put in the background to avoid hight impact on
+ *          the ILIAS system.
+ *
+ *          Whenever packing a bucket, you start with an Input (IO-interface) wich will be passed to the first task. your task will provide an output
+ *          (IO-interface again) wich will be passed to the next Task, ...
+ *
+ *          Bucket (IO -> Task -> IO -> Task)
```

Comment:
Yes it jumps right into your face after a while. I think that's why i subconciously named the value containers IO. What we would really want is pretty much the IO Monad from haskell together with its "do" syntax sugar. And it should be serializable. 

The pipe was enough for unix ;-) But I agree, it's just really hard to think of an interface that is easy to use and complete enough for all possible. Maybe:
$bucket = new Bucket();
$add = new AddJob();
$a = new IntegerValue(1);
$b = new IntegerValue(2);
$operation->addTask("variable1", $add, [$a, $b]);
$operation->addTask("returnValue", $add, [$a, "variable1"]);
$operation->setReturnValue("returnValue");
with 
/**
* @param $variableName string
* @param $task Task
* @param $parameters (string|Value)[] If string => variable if Value then constant.
**/
addTask($variableName, $task, $parameters);